test/unit/token/row-token-parser-test.js

```diff
@@ -0,0 +1,981 @@
+  buffer.writeString(value.slice(0, 3));
+  buffer.writeUInt32LE(3);
+  buffer.writeString(value.slice(3, 6));
+  buffer.writeUInt32LE(0);
+  //console.log(buffer.data)
+
+  var parser = new Parser({ token() {} }, colMetaData, options);
+  parser.write(buffer.data);
+  var token = parser.read();
+  //console.log(token)
+
+  test.strictEqual(token.columns.length, 1);
+  test.strictEqual(token.columns[0].value, value);
+  test.strictEqual(token.columns[0].metadata, colMetaData[0]);
+
+  test.done();
+};
+
+module.exports.varCharMaxKnownLength = function(test) {
+  var colMetaData = [
+    {
+      type: dataTypeByName.VarChar,
+      dataLength: 65535,
+      collation: {
+        codepage: undefined
+      }
+    }
+  ];
+  var value = 'abcdef';
+
+  var buffer = new WritableTrackingBuffer(0, 'ascii');
+  buffer.writeUInt8(0xd1);
+  buffer.writeUInt64LE(value.length);
+  buffer.writeUInt32LE(3);
+  buffer.writeString(value.slice(0, 3));
+  buffer.writeUInt32LE(3);
+  buffer.writeString(value.slice(3, 6));
+  buffer.writeUInt32LE(0);
+  //console.log(buffer.data)
+
+  var parser = new Parser({ token() {} }, colMetaData, options);
+  parser.write(buffer.data);
+  var token = parser.read();
+  //console.log(token)
+
+  test.strictEqual(token.columns.length, 1);
+  test.strictEqual(token.columns[0].value, value);
+  test.strictEqual(token.columns[0].metadata, colMetaData[0]);
+
+  test.done();
+};
+
+module.exports.varCharMaxWithCodepage = function(test) {
+  var colMetaData = [
+    {
+      type: dataTypeByName.VarChar,
+      dataLength: 65535,
+      collation: {
+        codepage: 'WINDOWS-1252'
+      }
+    }
+  ];
+  var value = 'abcdéf';
+
+  var buffer = new WritableTrackingBuffer(0, 'ascii');
+  buffer.writeUInt8(0xd1);
+  buffer.writeUInt64LE(value.length);
+  buffer.writeUInt32LE(3);
+  buffer.writeString(value.slice(0, 3));
+  buffer.writeUInt32LE(3);
+  buffer.writeString(value.slice(3, 6));
+  buffer.writeUInt32LE(0);
+  //console.log(buffer.data)
+
+  var parser = new Parser({ token() {} }, colMetaData, options);
+  parser.write(buffer.data);
+  var token = parser.read();
+  //console.log(token)
+
+  test.strictEqual(token.columns.length, 1);
+  test.strictEqual(token.columns[0].value, value);
+  test.strictEqual(token.columns[0].metadata, colMetaData[0]);
+
+  test.done();
+};
+
+module.exports.varCharMaxKnownLengthWrong = function(test) {
+  var colMetaData = [
+    {
+      type: dataTypeByName.VarChar,
+      dataLength: 65535
+    }
+  ];
+  var value = 'abcdef';
+
+  var buffer = new WritableTrackingBuffer(0, 'ascii');
+  buffer.writeUInt8(0xd1);
+  buffer.writeUInt64LE(value.length + 1);
+  buffer.writeUInt32LE(3);
+  buffer.writeString(value.slice(0, 3));
+  buffer.writeUInt32LE(3);
+  buffer.writeString(value.slice(3, 6));
+  buffer.writeUInt32LE(0);
+  //console.log(buffer.data)
+
+  try {
+    var parser = new Parser({ token() {} }, colMetaData, options);
+    parser.write(buffer.data);
+    /* eslint-disable */
```

Comment:
You can get rid of this by changing:
<code>var token = parser.read();</code>
to
<code>parser.read();</code>